hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java

```diff
@@ -369,6 +373,9 @@ private synchronized void uploadCurrentBlock(boolean isLast)
    */
   @Retries.RetryTranslated
   private void initMultipartUpload() throws IOException {
+    if (!isMultipartAllowed){
+      return;
```

Comment:
throw Exception.

Reply:
Why do we need to throw exception here, since this just meant to skip the initialisation? 
cc: @mehakmeet @steveloughran

Reply:
good q. we should throw it because if something does get as far as calling it, something has gone very wrong.

I would just use 
```
Preconditions.checkState(!isMultipartEnabled, "multipart upload is disabled")
```